src/lib.rs

```diff
@@ -128,23 +149,25 @@ impl Drop for IRMetalLibBinary {
 }
 
 impl IRMetalLibBinary {
-    pub fn new(compiler: &IRCompiler) -> Result<IRMetalLibBinary, Box<dyn std::error::Error>> {
+    pub fn new(compiler: &IRCompiler) -> IRMetalLibBinary {
         unsafe {
-            let me = compiler.funcs.IRMetalLibBinaryCreate();
-            Ok(Self {
-                me,
+            let me = NonNull::new(compiler.funcs.IRMetalLibBinaryCreate())
+                .expect("Failed to create empty IRMetalLibBinary");
+            Self {
+                me: me.as_ptr(),
                 funcs: compiler.funcs.clone(),
-            })
+            }
         }
     }
 
     pub fn get_byte_code(&self) -> Vec<u8> {
         let size_in_bytes = unsafe { self.funcs.IRMetalLibGetBytecodeSize(self.me) };
         let mut bytes = vec![0u8; size_in_bytes];
```

Comment:
We could leave the mem uninitialized by calling `Vec::with_capacity()`, and assume they're safely initialized by calling `unsafe { set_size(witten) }` when we know how many bytes the API promised to have written.


Example:
https://github.com/ash-rs/ash/blob/660553c9184997c805c5a9f990395eab6d5e8dd4/ash/src/prelude.rs#L55